graylog2-server/src/main/java/org/graylog/events/rest/EventNotificationsResource.java

```diff
@@ -129,13 +134,15 @@ public NotificationDto update(@ApiParam(name = "notificationId") @PathParam("not
     @ApiOperation("Delete a notification")
     @AuditEvent(type = EventsAuditEventTypes.EVENT_NOTIFICATION_DELETE)
     public void delete(@ApiParam(name = "notificationId") @PathParam("notificationId") @NotBlank String notificationId) {
+        checkPermission(RestPermissions.EVENT_NOTIFICATIONS_DELETE);
         resourceHandler.delete(notificationId);
     }
 
     @GET
     @Path("/legacy/types")
     @ApiOperation("List all available legacy alarm callback types")
     public Response legacyTypes() {
+        checkPermission(RestPermissions.EVENT_NOTIFICATIONS_READ);
```

Comment:
We can use the `@RequiresPermssions` annotation here.

Reply:
On a second thought, I think we might want to leave that one without any permissions check.
If we want to allow a user to modify a single notification, he'd need that to use the UI.
Also I don't see much harm exposing that.